ros2action/test/test_echo.py

```diff
@@ -109,181 +277,28 @@ def test_echo(self):
-
-            for msg_block in expected_output:
-
-                # Find the position where the first message of the message block appears.
-                matched_begin_lines = []
-                if isinstance(msg_block[0], str):
-                    for index, line in enumerate(output_lines):
-                        if msg_block[0] == line:
-                            matched_begin_lines.append(index)
-
-                if hasattr(msg_block[0], 'match'):
-                    for index, line in enumerate(output_lines):
-                        if msg_block[0].match(line):
-                            matched_begin_lines.append(index)
-
-                filtered_matched_begin_lines = matched_begin_lines
-                if matched_begin_lines:
-                    for line in matched_begin_lines:
-                        if len(msg_block) > len(output_lines) - line:
-                            filtered_matched_begin_lines.remove(line)
-
-                    if not filtered_matched_begin_lines:
-                        return False
-                else:
-                    return False
-
-                # filter_matched_begin_lines include all positions where the first message appears.
-
-                msg_block_matched = False
-                for line in filtered_matched_begin_lines:
-                    matched_index = -1
-                    for index, (expected_line, output_line) in enumerate(
-                        zip(msg_block, output_lines[line:]), start=1
-                    ):
-                        if isinstance(expected_line, str):
-                            if expected_line != output_line:
-                                break
-                            else:
-                                matched_index = index
-                                continue
-                        if hasattr(expected_line, 'match'):
-                            if not expected_line.match(output_line):
-                                break
-                            else:
-                                matched_index = index
-                                continue
-                    # A complete match has been found, skipping the rest of the detection.
-                    if (matched_index == len(msg_block)):
-                        msg_block_matched = True
-                        break
-
-                # If no message block is matched throughout the entire output.
-                if not msg_block_matched:
-                    return False
+        with self.launch_action_command(arguments=echo_arguments) as action_command:
+            assert action_command.wait_for_output(
+                functools.partial(
+                    match_expected_output,
+                    expected_output=EXPECTED_OUTPUT,
+                ),
+                timeout=10,
+            )
+        assert action_command.wait_for_shutdown(timeout=10)
```

Comment:
I needed to adjust these a bit. because that is not aligned with comment, and zenoh requires longer wait timeout, otherwise it fails here.